src/components/searchForm.jsx

```diff
@@ -1,40 +1,47 @@
 import React from 'react';
 import InputWithAutocomplete from './inputWithAutocomplete';
+import * as locationsMap from '../../locations/ukraine.json';
 
 class SearchForm extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
             inputText: '',
-            isSubmitable: false
+            isSubmitable: false,
+            wasChanged: false,
+            locationsNames: Object.keys(locationsMap)
         }
-
         this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
     }
 
     handleChange(value) {
         this.setState({
             inputText: value,
-            isSubmitable: this.isValid(value)
+            isSubmitable: this.isSubmitable(value),
+            wasChanged: true
         });
     }
 
-    isValid(part) {
-        var validValues = (part === '') ? [] : this.props.locations.filter((location) => location.name.toUpperCase().startsWith(part.toUpperCase())).slice(0, 10).find((hint) => hint.name === part);
-        return validValues !== undefined;
+    handleSubmit(event) {
+        event.preventDefault();
+        alert('Submitted');
     }
 
-    render() {
-        let validationMessage = (this.state.isSubmitable || this.state.inputText === '') ? null : <p className='validation-message'>Input value is no Valid</p>
+    isSubmitable(part) {
+        return locationsMap.hasOwnProperty(part);
+    }
 
+    render() {
         return (
             <form className='search-form'>
-                <InputWithAutocomplete onChange={this.handleChange} value={this.state.inputText} locations={this.props.locations} />
-
-                <button className='search-button' type='submit' disabled={!this.state.isSubmitable}>Show Weather</button>
-                {validationMessage}
+                <InputWithAutocomplete onChange={this.handleChange} value={this.state.inputText} options={this.state.locationsNames} />
+                <button onClick={this.handleSubmit} className='search-button' type='submit' disabled={!this.state.isSubmitable}>Show Weather</button>
```

Comment:
Now I check isSubmitable flag in handler of form submit, but I left checking of it in disabled property off button